File: Ruby/Primes.rb
PRIME_COUNT = 500_000
primes = [2]

def prime?(num)
  max_factor = Math.sqrt(num).floor
  (2..max_factor).each do |factor|
    return false if (num % factor).zero?
  end
  true
end

start_time = Time.now
candidate = 3
while primes.size < PRIME_COUNT do
  primes << candidate if prime?(candidate)
  candidate += 2
end
elapsed = Time.now - start_time

puts "last prime found: #{primes[-1]}"
puts "elapsed: #{elapsed}"

File: Python/Primes.py
import datetime, math

PRIME_COUNT = 5_000
primes = [2]

def is_prime(num):
    max_factor = math.floor(math.sqrt(num))
    for factor in range(2, max_factor + 1):
        if num % factor == 0:
            return False
    return True

start_time = datetime.datetime.now()
candidate = 3
while len(primes) < PRIME_COUNT:
    if is_prime(candidate):
        primes.append(candidate)
    candidate += 2
elapsed = datetime.datetime.now() - start_time

print(f'last prime: {primes[-1]}')
print(f'elapsed: {elapsed}')
